evalml/data_checks/target_leakage_data_check.py

```diff
@@ -1,69 +1,72 @@
     """
 
-    def __init__(self, pct_corr_threshold=0.95, method="mutual"):
+    def __init__(self, pct_corr_threshold=0.95, method="mutual_info"):
         if pct_corr_threshold < 0 or pct_corr_threshold > 1:
             raise ValueError(
                 "pct_corr_threshold must be a float between 0 and 1, inclusive.",
             )
-        if method not in ["mutual", "pearson"]:
-            raise ValueError(f"Method '{method}' not in ['mutual', 'pearson']")
+        methods = CONFIG_DEFAULTS["correlation_metrics"]
+        if method not in methods:
+            raise ValueError(f"Method '{method}' not in {methods}")
+        if method == "all":
+            raise ValueError("Cannot use 'all' as the method")
         self.pct_corr_threshold = pct_corr_threshold
         self.method = method
 
-    def _calculate_pearson(self, X, y):
+    def _calculate_dependence(self, X, y):
         highly_corr_cols = []
-        X_num = X.ww.select(include=numeric_and_boolean_ww)
-        if (
-            y.ww.logical_type.type_string not in numeric_and_boolean_ww
-            or len(X_num.columns) == 0
-        ):
-            return highly_corr_cols
-        highly_corr_cols = [
-            label
-            for label, col in X_num.iteritems()
-            if abs(y.corr(col)) >= self.pct_corr_threshold
-        ]
-        return highly_corr_cols
-
-    def _calculate_mutual_information(self, X, y):
-        highly_corr_cols = []
-        for col in X.columns:
-            cols_to_compare = X.ww[[col]]
-            cols_to_compare.ww[str(col) + "y"] = y
-            mutual_info = cols_to_compare.ww.mutual_information()
-            if (
-                len(mutual_info) > 0
-                and mutual_info["mutual_info"].iloc[0] > self.pct_corr_threshold
-            ):
-                highly_corr_cols.append(col)
+        X2 = X.ww.copy()
+        target_str = "target_y"
+        while target_str in list(X2.columns):
+            target_str += "_y"
+        X2.ww[target_str] = y
+        try:
+            dep_corr = X2.ww.dependence_dict(
+                measures=self.method,
+                target_col=target_str,
+            )
+        except KeyError:
+            # keyError raised when the target does not appear due to incompatibility with the metric, return []
+            return []
+        highly_corr_cols = sorted(
+            [
+                corr_info["column_1"]
+                for corr_info in dep_corr
+                if abs(corr_info[self.method]) >= self.pct_corr_threshold
+            ],
+            key=lambda x: X2.columns.tolist().index(x),
+        )
         return highly_corr_cols
 
     def validate(self, X, y):
-        """Check if any of the features are highly correlated with the target by using mutual information or Pearson correlation.
+        """Check if any of the features are highly correlated with the target by using mutual information, Pearson correlation, and/or Spearman correlation.
```

Comment:
does this method only work with MI, pearson and Spearman? Might need to clarify here.